src/util/structured_data.cpp

```diff
@@ -0,0 +1,218 @@
+// Copyright 2016-2020 Diffblue Limited. All Rights Reserved.
+
+#include "structured_data.h"
+#include "range.h"
+#include "string_utils.h"
+#include <algorithm>
+
+labelt::labelt(std::vector<std::string> components)
+{
+  auto to_lower_string = [](const std::string &s) -> std::string {
+    std::string lower_case = s;
+    std::for_each(
+      lower_case.begin(), lower_case.end(), [](char &c) { c = ::tolower(c); });
+    return lower_case;
+  };
+  this->components = make_range(components)
+                       .map(to_lower_string)
+                       .collect<std::vector<std::string>>();
+}
+std::string labelt::camel_case() const
+{
+  std::ostringstream output;
+  if(!components.empty())
+  {
+    output << *components.begin();
+    join_strings(
+      output, std::next(components.begin()), components.end(), "", capitalize);
+  }
+  return output.str();
+}
+std::string labelt::snake_case() const
+{
+  std::ostringstream output;
+  join_strings(output, components.begin(), components.end(), '_');
+  return output.str();
+}
+std::string labelt::kebab_case() const
+{
+  std::ostringstream output;
+  join_strings(output, components.begin(), components.end(), '-');
+  return output.str();
+}
+std::string labelt::pretty() const
+{
+  std::ostringstream output;
+  if(!components.empty())
+  {
+    const auto range =
+      make_range(components.begin(), std::next(components.begin()))
+        .map(capitalize)
+        .concat(make_range(std::next(components.begin()), components.end()));
+
+    join_strings(output, range.begin(), range.end(), ' ');
+  }
+  return output.str();
+}
+bool labelt::operator<(const labelt &other) const
+{
+  return camel_case() < other.camel_case();
```

Comment:
Could directly compare vectors instead

Reply:
Any reason to prefer this? Here I rely on the library implemented alphabetical sort of strings, which seems simpler than writing a sort for vector

Reply:
I think it's already implemented in the standard library for vectors

Reply:
One possible thing to be aware of, though not entirely sure if it applies here, and if so, whether it even matters or not... is that a lot of standard library sort routines on strings are actually locale sensitive... meaning you may/will get different sort orders on different platforms, or different users on the same platform if they change locale. If you want this to be at all stable in its sort order, you need to be aware of that. Locale sensitive sorting has caused issues with CI in the past (e.g. output from CBMC in different orders between CI and local runs)